internal/weapons/spear/crimsonmoonssemblance/crimsonmoonssemblance.go

```diff
@@ -0,0 +1,87 @@
+package crimsonmoonssemblance
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+const (
+	icdKey      = "crimsonmoonssemblance-icd"
+	icdDuration = 14 * 60
+)
+
+func init() {
+	core.RegisterWeaponFunc(keys.CrimsonMoonsSemblance, NewWeapon)
+}
+
+type Weapon struct {
+	Index int
+}
+
+func (w *Weapon) SetIndex(idx int) { w.Index = idx }
+func (w *Weapon) Init() error      { return nil }
+
+func NewWeapon(c *core.Core, char *character.CharWrapper, p info.WeaponProfile) (info.Weapon, error) {
+	w := &Weapon{}
+	refine := p.Refine
+
+	c.Events.Subscribe(event.OnEnemyHit, func(args ...interface{}) bool {
+		ae := args[1].(*combat.AttackEvent)
+
+		if ae.Info.AttackTag != attacks.AttackTagExtra {
+			return false
+		}
+
+		if c.Player.Active() != char.Index {
+			return false
+		}
+
+		if char.StatusIsActive(icdKey) {
+			return false
+		}
+
+		char.AddStatus(icdKey, icdDuration, true)
+		char.ModifyHPDebtByRatio(0.25)
+
+		return false
+	}, fmt.Sprintf("crimsonmoonssemblance-hit-%v", char.Base.Key.String()))
+
+	c.Events.Subscribe(event.OnHPDebt, func(args ...interface{}) bool {
```

Comment:
I *think* this entire block should just be an AddStatMod, with the current debt percent being a check inside the `Amount` func.

This code here would cause the mod to be updated only when hp debt changes. While the result is the same (since event would have to be triggered for BOL to reduce), it seems a little inefficient to re-create a slice and re-add a mod every time BOL changes. Better to just add a permanent mod and check the BOL amount each time ?